WBooks/Views/BookIndex/BookIndexCell.swift

```diff
@@ -37,22 +42,19 @@ final internal class BookIndexCell: UITableViewCell, NibLoadable {
 
 private extension BookIndexCell {
     
-    func configureImage(urlBook: URL?) {
-        guard let url = urlBook else { return }
-        URLSession.shared.dataTask(with: url) { (data, response, error) in
-            if error != nil {
-                print("Failed fetching image \(error.debugDescription)")
-                return
-            }
-            
-            guard let response = response as? HTTPURLResponse, response.statusCode == 200 else {
-                print("Not a proper HTTPURLResponse or statusCode")
-                return
-            }
-            
-            DispatchQueue.main.async {
-                self.bookImage.image = UIImage(data: data!)
+    func configureImage(url: URL?) {
+        
+        guard let imageURL = url else { return }
+        imageFetcher.fetchImage(imageURL).startWithResult {[unowned self] result in
```

Comment:
You can avoid the explicit `DispatchQueue` using `SignalProducer`'s [`observe(on:)`](http://reactivecocoa.io/reactiveswift/docs/latest/Structs/SignalProducer.html#/s:13ReactiveSwift14SignalProducerV7observeACyxq_GAA9Scheduler_p2on_tF) method:

```swift
imageFetcher.fetchImage(imageURL)
  .observe(on: UIScheduler())
  .startWithResult { [onwoned self] result in 
    switch result {
    case .success(let image):
      self.bookImage.image = image
    case .failure(let error):
      print(error)
    }
  }
```

More on schedulers [here](https://github.com/ReactiveCocoa/ReactiveSwift/blob/master/Documentation/FrameworkOverview.md#schedulers), [here](https://github.com/ReactiveCocoa/ReactiveSwift/blob/master/Documentation/APIContracts.md#observe-events-on-a-known-scheduler) and [here](https://github.com/ReactiveCocoa/ReactiveSwift/blob/master/Documentation/APIContracts.md#events-are-sent-synchronously-by-default)

Reply:
Also this solution stills has the problem of maybe displaying the wrong image if the cell gets reused, you need to cancel the signal producer if the cell gets reused. Read about disposables [here](https://github.com/ReactiveCocoa/ReactiveSwift/blob/master/Documentation/FrameworkOverview.md#disposables)

```swift
final internal class BookIndexCell: UITableViewCell, NibLoadable {

   var disposable: Disposable?

   override func prepareForReuse() {
       disposable?.dispose()
       disposable = .none
   }

}

private extension BookIndexCell {
    
    func configureImage(url: URL?) {
        guard let imageURL = url else { return }
        disposable = imageFetcher.fetchImage(imageURL)
          .observe(on: UIScheduler())
          .startWithResult { [onwoned self] result in 
              switch result {
              case .success(let image):
                  self.bookImage.image = image
              case .failure(let error):
                  print(error)
              }
          }
    }

}
```